core/drivers/imx_wdog.c

```diff
@@ -124,28 +112,38 @@ static TEE_Result imx_wdog_base(vaddr_t *wdog_vbase)
 	}
 
 	/* search the first usable wdog */
-	for (i = 0; i < ARRAY_SIZE(wdog_path); i++) {
-		off = fdt_path_offset(fdt, wdog_path[i]);
-		if (off < 0)
-			continue;
-
-		st = _fdt_get_status(fdt, off);
-		if (st & DT_STATUS_OK_SEC)
+	for (i = 0; i < ARRAY_SIZE(dt_wdog_match_table); i++) {
+		match = dt_wdog_match_table[i];
+		off = 0;
+		while (off >= 0) {
+			off = fdt_node_offset_by_compatible(fdt, off, match);
+			if (off > 0) {
+				st = _fdt_get_status(fdt, off);
+				if (st & DT_STATUS_OK_SEC) {
+					DMSG("Wdog found at %u", off);
+					found_off = off;
+					break;
+				}
+			}
+		}
```

Comment:
For this loop, I would do as the following:
```
	for (i = 0; i < ARRAY_SIZE(dt_match_table); i++) {
		off = fdt_node_offset_by_compatible(fdt, 0, dt_match_table[i]);

		for (; off != -FDT_ERR_NOTFOUND;
		     off = fdt_node_offset_by_compatible(fdt, off,
							 dt_match_table[i])) {
			if (_fdt_get_status(fdt, off) != DT_STATUS_DISABLED)
				break;
		}

		if (off >= 0)
			break;
	}
```

Reply:
I find the `for` loop to be readable, but hard to understand. Even with the additional variables used in my version, I don't think there should be a code size difference, I can check if you want me to. Otherwise I'd prefer my version.

Reply:
Have you looked?
I'm fine going with your version anyway

Reply:
No, I haven't checked that yet, but would prefer my version.